moxy/org.eclipse.persistence.moxy/src/test/java/org/eclipse/persistence/testing/moxy/unit/jaxb/compiler/builder/TransformerPropertyBuilderTestCase.java

```diff
@@ -11,225 +11,87 @@
+        Map<String, Map<String, Source>> properties = new HashMap<>();
+        properties.put(JAXBContextProperties.OXM_METADATA_SOURCE, metadataSourceMap);
+        Map<String, XmlBindings> bindings = JAXBContextFactory.getXmlBindingsFromProperties(properties, Thread.currentThread().getContextClassLoader());
+        JavaModelInputImpl jModelInput = new JavaModelInputImpl(DOMAIN_CLASSES, new JavaModelImpl(new JaxbClassLoader(Thread.currentThread().getContextClassLoader(), DOMAIN_CLASSES)));
+        Helper helper = new Helper(jModelInput.getJavaModel());
+        Generator generator = new Generator(jModelInput, bindings, Thread.currentThread().getContextClassLoader(), "", false);
+        TypeInfo typeInfo = generator.getAnnotationsProcessor().getTypeInfos().get(EMPLOYEE_CLASS_NAME);
+        Property normalHoursProperty = typeInfo.getProperties().get(XML_TRANSFORMATION_PROPERTY_NAME);
+        ChildSchemaGenerator childSchemaGenerator = new ChildSchemaGenerator(helper);
+        TransformerPropertyBuilder transformerPropertyBuilder = childSchemaGenerator.getTransformerPropertyBuilder(normalHoursProperty, typeInfo);
+        // Indirect call of org.eclipse.persistence.jaxb.compiler.builder.TransformerPropertyBuilder.getPropertyName(...) and org.eclipse.persistence.jaxb.compiler.builder.TransformerPropertyBuilder.buildPropertyFromTransformer(...)
+        // Indirect call of org.eclipse.persistence.jaxb.compiler.builder.TransformerPropertyBuilder.getTransformerJavaClass(...) with booth options writeTransformer.isSetTransformerClass() true,false
+        List<Property> props = transformerPropertyBuilder.buildProperties();
+        assertEquals(2, props.size());
+        // Verification of result of org.eclipse.persistence.jaxb.compiler.builder.TransformerPropertyBuilder.getPropertyName(...)
+        // Verify property names
+        assertEquals("start-time", props.get(0).getPropertyName());
+        assertEquals("end-time", props.get(1).getPropertyName());
+        // Verification of result of org.eclipse.persistence.jaxb.compiler.builder.TransformerPropertyBuilder.getReturnTypeFromTransformer(...)
+        // Verify property types
+        assertEquals("java.lang.String", props.get(0).getType().getName());
+        assertEquals("java.lang.String", props.get(1).getType().getName());
+
+        assertEquals("org.eclipse.persistence.testing.jaxb.externalizedmetadata.mappings.xmltransformation.NormalHoursAttributeTransformer", normalHoursProperty.getXmlTransformation().getXmlReadTransformer().getTransformerClass());
+        assertEquals("org.eclipse.persistence.testing.jaxb.externalizedmetadata.mappings.xmltransformation.StartTimeTransformer", normalHoursProperty.getXmlTransformation().getXmlWriteTransformer().get(0).getTransformerClass());
+        assertEquals("java.lang.String", normalHoursProperty.getGenericType().getName());
```

Comment:
when working with class names, it is usually better to use sth like `String.class.getName()` or `NormalHoursAttributeTransformer.class.getName()`. The benefit becomes visible once someone tries to refactor the code or even simply find usages of given class. Not all IDEs do have ability to search through string values...